src/base/bittorrent/tracker.cpp

```diff
@@ -107,7 +107,7 @@ namespace
         case QAbstractSocket::IPv6Protocol: {
                 const Q_IPV6ADDR ipv6 = addr.toIPv6Address();
                 QByteArray ret;
-                for (int i = (sizeof(ipv6.c) - 1); i >= 0; --i)
```

Comment:
Could you explain a bit more about this change?

Reply:
See http://www.bittorrent.org/beps/bep_0023.html

> Both address and port use network-byte order.

and https://en.wikipedia.org/wiki/Endianness#Networking

>(...) the term _network order_, meaning the order of transmission for bits and bytes over the wire in network protocols. Among others, the historic [RFC 1700](https://tools.ietf.org/html/rfc1700) (also known as Internet standard STD 2) has defined the network order for protocols in the Internet protocol suite to be big-endian, hence the use of the term "network byte order" for big-endian byte order.

This basically means "most significant byte appears first" which is what this change did. Previously, this function was inverting the order of all bytes, effectively encoding the address as little endian i.e. "least significant byte appears first".

Reply:
Now I recall I didn't have the environment to test IPv6 related code at that time. Good that you fixed it.